ml_peg/app/utils/register_callbacks.py

```diff
@@ -50,24 +54,37 @@ def update_summary_table(
             Value of selected tab. Parameter unused, but required to register Input.
         stored_weights
             Stored summary weights dictionary.
+        selected_models
+            Currently selected MLIPs from the global model filter.
         stored_scores
             Stored scores for table scores.
         summary_data
-            Data from summary table to be updated.
+            Latest computed summary table rows.
 
         Returns
         -------
-        list[dict]
-            Updated summary table data.
+        tuple[list[dict], list[dict], list[dict]]
+            Filtered table data, style, and cached unfiltered rows.
         """
+        summary_rows = deepcopy(summary_data) if summary_data else []
+        if not summary_rows:
+            raise PreventUpdate
+
         # Update table from stored scores
         if stored_scores:
-            for row in summary_data:
+            for row in summary_rows:
                 for tab, values in stored_scores.items():
-                    row[tab] = values[row["MLIP"]]
+                    row[tab] = values.get(row["MLIP"])
+
+        full_rows, _ = update_score_style(summary_rows, stored_weights)
+        filtered_rows = filter_rows_by_models(full_rows, selected_models)
+        if filtered_rows:
+            filtered_scores = calc_metric_scores(filtered_rows)
```

Comment:
Why would `filter_rows_by_models` not filter the scores too? Why do we need to recalculate them here and every other time `filter_rows_by_models` is called?